### PR TITLE
[Merged by Bors] - feat: port Algebra.Order.Interval

### PR DESCRIPTION
**Pull request by int-y1**
---

todo:

* [x] `NonemptyInterval.subtractionCommMonoid` errors with `sub_eq_add_neg` and `neg_add_rev`
* [x] `NonemptyInterval.divisionCommMonoid` same thing

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by xroblot** [comment]
Pushed some fixes. It was necessary to apply two `ext` lemmas to recover the output of the `ext` from Mathlib3. 

Beware when naming instances that there is already a namespace so it is not necessary to add it to the instance name.

**Comment by Komyyy** [comment]
I'm working on the `positivity` extension.

File: Mathlib/Algebra/Order/Interval.lean [hunk 1]
```diff
@@ -0,0 +1,783 @@
...
+--   | q(Interval.length $(s)) => nonnegative <$> mk_app `interval.length_nonneg [s]
+--   | e =>
+--     pp e >>=
+--       fail ∘
+--         format.bracket "The expression `"
+--           "` isn't of the form `nonempty_interval.length s` or `interval.length s`"
+-- #align tactic.positivity_interval_length tactic.positivity_interval_length
+
+-- end Tactic
+
```
Comment (semorrison) [inline Mathlib/Algebra/Order/Interval.lean:751]: Remove now that you've written the extension.

**Comment by semorrison** [comment]
bors d+

**Pull request closed without merging**